client/src/pages/Shop/index.tsx

```diff
@@ -16,6 +19,12 @@ import ShopComponent from './styles'
 export const Shop = observer(() => {
     const imageSources: string[] = [Images.Group, Images.Welding, Images.Stock]
 
+    useEffect(() => {
+        if (userStore.isAuth) {
+            basketStore.fetchBasket(userStore.user!.id).then(() => basketStore.setStatus(StatusEnum.initial))
+        }
+    }, [userStore.isAuth])
```

Comment:
You have the same useEffect above. For cases like this we can create custom hooks to reduce amount of duplicated code. You can create custom hook called useFetchBasket and pass there an array of dependencies (it's just that in the example above you don't pass anything and here you pass isAuth)